src/libpmemobj/memops.c

```diff
@@ -397,14 +397,27 @@ operation_add_buffer(struct operation_context *ctx,
 
 	size_t curr_size = MIN(real_size, ctx->ulog_curr_capacity);
 	size_t data_size = curr_size - sizeof(struct ulog_entry_buf);
+	size_t entry_size = ALIGN_UP(curr_size, CACHELINE_SIZE);
+
+	size_t next_entry_offset = ctx->ulog_curr_offset + entry_size;
+	struct ulog_entry_base *next_entry = NULL;
+	if (next_entry_offset == ctx->ulog_curr_capacity) {
+		struct ulog *u = ulog_next(ctx->ulog_curr, ctx->p_ops);
```

Comment:
I wonder what will happen when it is the last entry in the log and the next ulog is NULL?
Then we do not clobber any entry. I guess (but I'm not sure) that's ok because we validate until next ulog is not a null - am I right?

Reply:
yes